python/lsst/pipe/tasks/postprocess.py

```diff
@@ -532,6 +574,21 @@ def inputDataset(self):
     def ConfigClass(self):
         raise NotImplementedError('Subclass must define "ConfigClass" attribute')
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        if self.config.functorFile:
+            self.log.info('Loading tranform functor definitions from %s',
+                          self.config.functorFile)
+            self.funcs = CompositeFunctor.from_file(self.config.functorFile)
+            self.funcs.update(dict(PostprocessAnalysis._defaultFuncs))
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        result = self.run(parq=inputs['inputCatalog'], funcs=self.funcs,
+                          dataId=outputRefs.outputCatalog.dataId)
```

Comment:
I'm not sure if using dataId in this way is really kosher in Gen3.  Specifically, `filter` is now `physical_filter`.  So in this case, I think that this won't work correctly.  I'm trying to follow how `dataId` gets used downstream, if there's anything else that needs to come from it; if it's just returning the dataId in the struct that seems like a gen2 necessity not gen3 (though I could be wrong).

What I would suggest is specifically pulling out `physical_filter` for gen3 (and `filter` for gen2), and having that as a separate keyword which seems special and necessary.  And then I think the `dataId` can be set only in the gen2 code run.  Though I may be missing one of the uses here.

Reply:
Ah right, because the regular dataId depends on what you pass on the command-line. OK I changed it `outputRefs.outputCatalog.dataId.full` so that it'll at least be the same keys are being added to the table every time.  Technically, all this info is redundant: everything will eventually be looked up with the "cccVisitId."  I've been adding these columns because we don't have a corresponding `ccdVisit` table yet to look up the full ccd info given a ccdVisitId. 

Passing an optional dataId for logging was kosher last I checked.